packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,632 @@
+      if (vCalValue.hasOwnProperty('start')) {
+        start = vCalValue.start
+        end = vCalValue.end
+      } else {
+        start = vCalValue
+      }
+
+      // Set emitted value when v-calendar selection is made.
+      // The `timePeriodsKey` param only applies to relative timeframes,
+      // not `v-calendar` selections; however, this keeps the object "shape" consistent.
+      selectedRange.value = {
+        start: start ? new Date(start) : '',
+        end: end ? new Date(end) : '',
+        timePeriodsKey: ''
+      }
+    }
+
+    /**
+     * Updates both the input field value, and the full time frame readout
+     * when a relative time frame button is clicked
+     * @param {*} timeframe
+     */
+    const changeRelativeTimeframe = (timeframe) => {
+      selectedTimeframe.value = timeframe
+
+      // Format the start/end values as human readable date
+      const start = selectedTimeframe.value.start()
+      const end = selectedTimeframe.value.end()
+
+      // Set value to be emitted when relative time frame clicked
+      selectedRange.value = {
+        start: new Date(start),
+        end: new Date(end),
+        timePeriodsKey: selectedTimeframe.value.key
+      }
+
+      fullRangeDisplay.value = formatDisplayDate({ start, end })
+    }
+
+    /**
+     * Clears any previously made choices, and emits the result of this action
+     * back to the parent.
+     */
+    const clearSelection = () => {
+      selectedCalendarRange.value = null
+      abbreviatedDisplay.value = props.placeholder
+      fullRangeDisplay.value = ''
+      selectedRange.value = { start: '', end: '', timePeriodsKey: '' }
+
+      if (hasTimePeriods) {
+        selectedTimeframe.value = props.timePeriods[0]
+      }
+
+      // If the calendar has focus, and `range` is set to false, we are displaying
+      // a single date / time picker; therefore, need to emit an empty string.
+      //
+      // Else, we are displaying a date / time range (relative timeframes or calendar range)
+      if (!props.range && showCalendar) {
+        emit('changed', '')
+      } else {
+        emit('changed', selectedRange.value)
+      }
+    }
+
+    /**
+     * Displays selected date/time/range as a human readable string.
+     * The date formatting string is dynamically determined based on
+     * the current mode of the instance (Custom vs Relative)
+     * @param {*} range A set of `start` and `end` Unix timestamps
+     */
+    const formatDisplayDate = (range) => {
+      const { start, end } = range
+      let fmtStr = 'PP'
+
+      // Determines the human timestamp readout format string; subject to change
+      if (!hasCalendar.value && hasTimePeriods.value) {
+        fmtStr = 'PP hh:mm a'
+      } else if (props.mode === 'date') {
+        fmtStr = 'PP'
+      } else if (props.mode === 'time') {
+        fmtStr = 'PP hh:mm a'
+      } else if (props.mode === 'dateTime') {
+        fmtStr = 'PP hh:mm a'
```

Comment:
question: would it make sense to allow the user to pass in the format they want displayed, or do we keep as-is where the component controls it? 

No wrong answers here, just thinking out loud.

Reply:
If they pass things in, the onus is on them not to break things.  Whereas us setting it internally provides some sense of security :)

I'm okay with this, if it makes the Kongponent have less lines of code. Will add this idea in 
https://konghq.atlassian.net/browse/MA-848 so we can discuss further.

Reply:
Yeah maybe future-state -- fine for now